core/services/relay/evm/evm.go

```diff
@@ -218,21 +218,43 @@ func NewRelayer(ctx context.Context, lggr logger.Logger, chain legacyevm.Chain,
 	}
 
 	wCfg := chain.Config().EVM().Workflow()
-	// Initialize write target capability if configuration is defined
-	if wCfg.ForwarderAddress() != nil && wCfg.FromAddress() != nil {
-		if wCfg.GasLimitDefault() == nil {
-			return nil, fmt.Errorf("unable to instantiate write target as default gas limit is not set")
+
+	// Do not register write target without required configuration
+	if wCfg.ForwarderAddress() == nil || wCfg.GasLimitDefault() == nil {
+		return relayer, nil
+	}
+
+	fromAddress := wCfg.FromAddress()
+
+	// If fromAddress is not specified, default to the eth key if there is only one
+	if fromAddress == nil {
+		// Neither fromAddress nor fromAddressIndex is specified, return without registering write target
+		if wCfg.FromAddressIndex() == nil {
+			return relayer, nil
 		}
 
-		capability, err := NewWriteTarget(ctx, relayer, chain, *wCfg.GasLimitDefault(), lggr)
-		if err != nil {
-			return nil, fmt.Errorf("failed to initialize write target: %w", err)
+		ethKeys, err2 := opts.CSAETHKeystore.Eth().GetAll(ctx)
```

Comment:
can you double check that this is stable wrt to addition over time? ie 0 is the first one that was added, not undefined or lexographically first?